src/Optical/Paraxial.jl

```diff
@@ -17,10 +17,10 @@ ParaxialLensConvexPoly(focaldistance, local_frame, local_polygon_points, local_c
 ```
 """
 struct ParaxialLens{T} <: Surface{T}
-    shape::Union{Rectangle{T},Ellipse{T},Hexagon{T},ConvexPolygon{T}}
```

Comment:
I don't think this will do what you want. I think this will define a ConvexPolygon whose first type argument is T. This might be what you want:

```
shape::Union{Rectangle{T},Ellipse{T},Hexagon{T},ConvexPolygon{N,T}} where{N}
```

Then for sure the T type argument to ConvexPolygon is in the right place.